### PR TITLE
[luci/import] Remove UnPack API from CircleReader class

### PR DESCRIPTION
**Pull request by m-bronnikov**
This commit removes UnPack API from CircleReader class.

ONE-DCO-1.0-Signed-off-by: Maksim Bronnikov <max120199@gmail.com>

-------------------------------

For: #7886
Draft: #7901

### CODE REVIEW

File: compiler/luci/import/include/luci/Import/CircleReader.h [hunk 1]
```diff
@@ -103,13 +103,6 @@ template <typename T> VectorWrapper<T> wrap(const flatbuffers::Vector<T> *vec)
  */
 class CircleReader
 {
-private: // unpack API
-  using CircleBuffers_t = std::vector<std::unique_ptr<circle::BufferT>>;
-  using CircleTensors_t = std::vector<std::unique_ptr<circle::TensorT>>;
-  using CircleOperators_t = std::vector<std::unique_ptr<circle::OperatorT>>;
-  using CircleOperatorCodes_t = std::vector<std::unique_ptr<circle::OperatorCodeT>>;
-  using CircleMetadata_t = std::vector<std::unique_ptr<circle::MetadataT>>;
```
Comment (m-bronnikov) [inline compiler/luci/import/include/luci/Import/CircleReader.h:111]: Remove type aliases.

File: compiler/luci/import/include/luci/Import/CircleReader.h [hunk 2]
```diff
@@ -120,20 +113,6 @@ class CircleReader
...
-  const CircleTensors_t &tensors() const { return _current_subgraph->tensors; }
-  const CircleOperators_t &operators() const { return _current_subgraph->operators; }
-  const std::vector<int32_t> &inputs() const { return _current_subgraph->inputs; }
-  const std::vector<int32_t> &outputs() const { return _current_subgraph->outputs; }
-  const std::string &name() const { return _current_subgraph->name; }
-  const circle::DataFormat &data_format() const { return _current_subgraph->data_format; }
-  const CircleMetadata_t &metadata() const { return _model->metadata; }
-
-  circle::BuiltinOperator builtin_code(const circle::OperatorT &op) const;
-  std::string opcode_name(const circle::OperatorT &op) const;
```
Comment (m-bronnikov) [inline compiler/luci/import/include/luci/Import/CircleReader.h:135]: Remove methods from *CircleReader.h* and *CircleReader.cpp*

File: compiler/luci/import/include/luci/Import/CircleReader.h [hunk 3]
```diff
@@ -155,9 +134,6 @@ class CircleReader
   bool select_subgraph(uint32_t subgraph);
 
 private:
-  std::unique_ptr<const circle::ModelT> _model;
-  const circle::SubGraphT *_current_subgraph{nullptr};
```
Comment (m-bronnikov) [inline compiler/luci/import/include/luci/Import/CircleReader.h:159]: Removed this 2 fields with usages.

**Comment by m-bronnikov** [comment]
I would like to remove helpers for UnPacked structures from CircleReader.h and CircleReader.cpp in next PR.

File: compiler/luci/import/src/CircleReader.cpp [hunk 4]
```diff
@@ -486,14 +457,12 @@ bool CircleReader::parse(const circle::Model *model)
 
 bool CircleReader::select_subgraph(uint32_t sgindex)
 {
-  if (_model->subgraphs.size() <= sgindex)
+  if (num_subgraph() <= sgindex)
```
Comment (seanshpark) [inline compiler/luci/import/src/CircleReader.cpp:460]: (optional) this seems side effect of removing

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**